service/src/main/java/com/codeforcommunity/requester/S3Requester.java

```diff
@@ -178,21 +181,76 @@ private static String validateBase64ImageAndUploadToS3(
    */
   public static String validateUploadImageToS3LucyEvents(String eventTitle, String base64Encoding)
       throws BadRequestImageException, S3FailedUploadException {
-    String fileName = getImageFileNameWithoutExtension(eventTitle);
+    String fileName = getFileNameWithoutExtension(eventTitle, "_thumbnail");
     return validateBase64ImageAndUploadToS3(fileName, externs.getDirPublic(), base64Encoding);
   }
 
   /**
-   * Removes special characters, replaces spaces, and appends "_thumbnail".
+   * Removes special characters, replaces spaces, and appends a suffix.
    *
    * @param eventTitle the title of the event.
+   * @param suffix the suffix to be appended
    * @return the String for the image file name (without the file extension).
    */
-  public static String getImageFileNameWithoutExtension(String eventTitle) {
+  public static String getFileNameWithoutExtension(String eventTitle, String suffix) {
     String title =
         eventTitle.replaceAll(
             "[!@#$%^&*()=+./\\\\|<>`~\\[\\]{}?]", ""); // Remove special characters
     return title.replace(" ", "_").toLowerCase()
-        + "_thumbnail"; // The desired name of the file in S3
+        + suffix; // The desired name of the file in S3
+  }
+
+  /**
+   * Validate the given string encoding of HTML and upload it to the user upload S3 bucket.
+   *
+   * @param name the desired name of the new file in S3 (without a file extension).
+   * @param directoryName the desired directory of the file in S3 (without leading or trailing '/').
+   * @param adminID the desired ID of the user uploading the HTML to S3.
+   * @param htmlContent the string encoding of the HTML to upload.
+   * @return null if the encoding fails validation and HTML file URL if the upload was successful.
```

Comment:
⛏️ Looks like we don't actually return `null` anywhere, so can change this line to say something like `@return the HTML file URL after a successful upload`?

Reply:
You are right-- I copied a bit too much from the upload image method. Thanks for catching that